be/src/util/compression/block_compression.cpp

```diff
@@ -742,7 +742,7 @@ class ZstdBlockCompression final : public BlockCompressionCodec {
         ZSTD_CCtx* ctx = context->ctx;
 
         [[maybe_unused]] faststring* compression_buffer = nullptr;
-        [[maybe_unused]] size_t max_len;
```

Comment:
why not put these variable in `if (use_compression_buffer) {` code block?

Reply:
max_len will be used lator
<img width="827" alt="image" src="https://user-images.githubusercontent.com/31178013/177126579-4f9fd832-bace-4dee-94d2-173419c5400e.png">

Reply:
I see that all usage happens in the if(use_compression_buffer) block. so I think it is OK to defined in this code block.